Break up Collections endpoint and extend WP_REST_Controller

obenland opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

Follow-up to https://github.com/Automattic/wordpress-activitypub/pull/1146.

## Proposed changes:
<!--- Explain what functional changes your PR includes -->
* Creates separate controllers for Replies and Moderators collections.
* Rewrites all three endpoints to build on WP_REST_Controller.
* Adds unit tests.

### Other information:

- [x] Have you written new tests for your changes, if applicable?

obenland committed:
Move file

obenland committed:
Break out endpoints and extend WP_REST_Controller

obenland committed:
Fixes

obenland committed:
Add tests

obenland committed:
Update tests

obenland committed:
Simplify if statement

obenland committed:
Reorder methods to make it easier to review

obenland committed:
Add prev to collection page

pfefferle committed:
Merge branch 'trunk' into update/collections-controller

pfefferle commented on includes/rest/class-replies-controller.php:
```diff
@@ -0,0 +1,171 @@
+		}
+
+		// Add ActivityPub Context.
+		$response['@context'] = Base_Object::JSON_LD_CONTEXT;
```
it does not really make a difference, but can we make `@context` the first item of the array, because of readability.

pfefferle committed:
fixes error with the replies collection

pfefferle committed:
Fix content-visibility in block editor (#1305)

* Fix content-visibility in block editor

The current version throws an error if the user wants to switch from a different visibility to `public`

* added changelog

* update/fix unittests

pfefferle committed:
cleanup empty values also on `add_post_meta`

pfefferle committed:
fix test

obenland committed:
Prepend context
